scipy/special/tests/test_basic.py

```diff
@@ -1990,6 +1990,8 @@ def test_factorial2(self):
         assert_array_almost_equal([105., 384., 945.],
                                   special.factorial2([7, 8, 9], exact=False))
         assert_equal(special.factorial2(7, exact=True), 105)
+        assert_equal(special.factorial2(0, exact=True), 1)
+        assert_equal(special.factorial2(-0.5, exact=True), 0)
```

Comment:
I wonder what the point of `exact` is for negative values since:

- it looks to me like negative integer values cannot be handled by the gamma function, so an approximation isn't possible to specify?
- for non-integer negative values the only option I see is to use the gamma function, so an exact calculation isn't really valid to specify?
- I don't see a solution for the negative factorial of even numbers?

It does seem to me like, as alluded to in the matching issue, adoption of the negative odd number handling used in the wikipedia section on this matter might make sense?

https://en.wikipedia.org/wiki/Double_factorial#Negative_arguments

As to whether an error should be raised for some of the permutations/situations I describe above, I may have to defer to `special` regulars.

Reply:
Not necessarily an important enough reason to keep the current functionality or extend it further to other negative odd integers, but I actually have some code that's uses `factorial2(-1)==1`.

It wouldn't be too hard to change, but it will be worth considering which direction will lead to more broken code (e.g making all negative integers return 0 or having the odd ones return other values).

Reply:
I tend to agree I don't think returning 0 for all negative values is the best way forwards.
The way that makes the most sense in my head is:

- Inputs of 0 & -1 return 1
- Negative odd integers are calculated using the method in the Wikipedia article
- Negative even integers and negative non-integers raise a `ValueError`

Reply:
I also notice that roughly the same code is present in `factorialk` which computes the kth order factorial (though only the exact version, as the other is not implemented). In principle, `factorial2(n)` could just be syntactic sugar for `factorialk(n,2)`

Another possibility is the alternative extension to all complex numbers (besides negative even integers or in general n%k==0 negative integers for `factorialk`).